similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -204,4 +211,41 @@ private long count(SolrQuery query)
         }
         return 0;
     }
+
+    private void appendGenesToQuery(PatientData<Map<String, String>> allGenes, StringBuilder q)
+    {
+        Collection<String> genesToSearch = new ArrayList<String>();
+        if (allGenes != null && allGenes.size() > 0 && allGenes.isIndexed()) {
+            for (Map<String, String> gene : allGenes) {
+                String geneName = gene.get("gene");
+                if (StringUtils.isBlank(geneName)) {
+                    continue;
+                }
+
+                geneName = geneName.trim();
+                String status = gene.get("status");
+                // Treat empty status as candidate
+                if (StringUtils.isBlank(status) || "solved".equals(status) || "candidate".equals(status)) {
+                    genesToSearch.add(geneName);
+                }
+            }
+        }
+        if (!genesToSearch.isEmpty()) {
+            String geneQuery = getQueryFromTerms(genesToSearch);
+            q.append(" solved_genes: " + geneQuery);
+            q.append(" candidate_genes: " + geneQuery);
```

Comment:
If you leave a space between `:` and `(`, then that is no longer a field specifier, so the query will actually look for the text `solved_genes` and each individual gene name in the fields specified in the qf parameter.